### DISCUSSION TITLE
master-details / grouped event stream view per trace?

### DISCUSSION THREAD

**Discussion by vmandic** [datalust/seq-tickets#1515] [Help] [answered]
Hi, 

I am really enjoying the SEQ application, we host it for test and prod separately, and test gets pretty loaded and noisy due to more envs. We use the signals feature to filter out per app and environment which removes the clutter a lot but it is still noisy when you wish to correlate across services through trace, a lot of logs fly in per a single request, eg. 5-10 logs get picked up. I know I could use Jaeger service for tracing eg. to visualize this better but I am curious if I can declutter the event stream view somehow. 

I am looking for a way to _declutter_ the log stream where I could only view a single request grouped by eg. TraceId property? I tried to write a query but my properties vary per apps a lot so its not that easy (or even a good approach?).

A query I tried was: `select TraceId, count(*) from stream group by TraceId limit 100` but this only gives out the count of logs per req. What I'd love to see is a kind of master-detail view where requests are grouped per trace\request ID, is this possible anyhow?

Thanks,
V.

(as this is my first post here I just want to say a huge thank you for making an incredible logger and log viewer)

**Comment by liammclennan** [comment, marked as answer]
Hi @vmandic,

Seq does not currently have a way of adding hierarchy to search results. 

You might try expanding an event, and selecting 'show as column' on your trace property

![image](https://user-images.githubusercontent.com/271514/158280806-e4c01a40-8fa2-40a6-946e-10bbf5049b44.png)

That will provide some visual indication of how events relate to traces.

![image](https://user-images.githubusercontent.com/271514/158280958-bb8a04f8-3043-4dda-bb5e-b10442d01b49.png)

**Reply by vmandic** [reply]
Okay that adds some verbosity to managing this. It would be great if multiple columns had labels. Would be lovely if  you could add this hierarchical viewing also, thanks.

**Reply by liammclennan** [reply]
You can have multiple columns, although it starts to look odd if there are too many. Thanks for the suggestion. We are always looking for ways to make the visualization more useful.
